src/platforms/evdev/libinput_device.cpp

```diff
@@ -338,6 +341,18 @@ mir::EventUPtr mie::LibInputDevice::convert_touch_frame(libinput_event_touch* to
             ++it;
     }
 
+    // Sanity check: Bogus panels are sending sometimes empty events that all point
+    // to (0, 0) coordinates. Detect those and drop the whole frame in this case.
+    // Also drop touch frames with no contacts inside
+    int emptyTouches = 0;
+    for_each(begin(contacts), end(contacts), [&](auto const& contact)
+    {
+        if (contact.x == 0 && contact.y == 0)
+            emptyTouches++;
+    });
```

Comment:
Lambda functions are nice, but they also have a certain performance and size cost: here I'd just do
```suggestion
    for (const auto &contact: contacts)
    {
        if (contact.x == 0 && contact.y == 0)
            emptyTouches++;
    }
```